classes/class.ilOpencastPageComponentPluginGUI.php

```diff
@@ -303,224 +230,136 @@ public function create(): void
     public function update(): void
     {
-        $form = $this->getForm();
-
-        $form->setValuesByPost();
+        $current_properties = $this->getProperties();
+        // retrieve new event id if changed
+        $event_id = $this->dic->http()->request()->getQueryParams()[self::PROP_EVENT_ID] ?? null;
+        if ($event_id !== null) {
+            $this->main_tpl->setOnScreenMessage('success', $this->translator->translate('msg_replaced'), true);
+            $current_properties[self::PROP_EVENT_ID] = $event_id;
+            $this->updateElement($current_properties);
+        }
 
-        if (!$form->checkInput()) {
-            $this->main_tpl->setContent($form->getHTML());
+        $edit = $this->getEditView();
+        $updated_properties = $edit->getUpdatedProperties($this->dic->http()->request());
 
+        if ($updated_properties === null || $event_id !== null) {
+            $this->main_tpl->setContent(
+                $this->dic->ui()->renderer()->render($edit->get())
+            );
             return;
         }
 
-        $properties = $this->getProperties();
-
-        $size = $form->getInput(self::POST_SIZE);
-        $properties[self::PROP_HEIGHT] = $size[self::PROP_HEIGHT];
-        $properties[self::PROP_WIDTH] = $size[self::PROP_WIDTH];
-        $properties[self::PROP_POSITION] = $form->getInput(self::PROP_POSITION);
-        $properties[self::PROP_RESPONSIVE] = $form->getInput(self::PROP_RESPONSIVE);
-        $properties[self::PROP_AS_LINK] = $form->getInput(self::PROP_AS_LINK);
-
+        $properties = array_merge($current_properties, $updated_properties);
         $this->updateElement($properties);
-
         $this->returnToParent();
     }
 
-    public function cancel(): void
-    {
-        $this->returnToParent();
-    }
+    // END performing commands
 
-    /**
-     * @param string $a_mode
-     * @param string $plugin_version
-     *
-     * @throws ilTemplateException
-     * @throws xoctException
-     */
-    public function getElementHTML($a_mode, array $a_properties, $plugin_version): string
-    {
-        try {
-            $event = $this->event_repository->find($a_properties[self::PROP_EVENT_ID]);
-        } catch (Exception) {
-            return $this->getExceptionHTML($a_properties);
-        }
-        $as_link = (bool) $a_properties[self::PROP_AS_LINK];
-        if (!$as_link && ($a_mode == self::MODE_PRESENTATION)) {
-            return $this->getIframeHTML($a_properties, $event);
-        }
-        return $this->getStandardElementHTML($a_mode, $a_properties, $event);
-    }
-
-    public function redirect(string $cmd): void
+    protected function addToolbar(): void
     {
-        $this->dic->ctrl()->setParameter($this, self::CUSTOM_CMD, $cmd);
-        $this->dic->ctrl()->redirect($this, self::CMD_INSERT);
+        $upload_button = ilLinkButton::getInstance();
+        $upload_button->setPrimary(true);
+        //        $this->dic->ctrl()->saveParameter($this, 'rtoken'); // TODO why???
```

Comment:
Again, please remove this or format the TODO: properly so that IDE plugins could read them!
```suggestion
        //        $this->dic->ctrl()->saveParameter($this, 'rtoken'); // TODO: why???
```

but I don't see any reason to keep it, so better to remove it!

Reply:
Will remove this as well